packages/core/src/initializer/index.ts

```diff
@@ -16,13 +16,34 @@ import { StrykerInquirer } from './stryker-inquirer.js';
 import { createInitializers } from './custom-initializers/index.js';
 import { GitignoreWriter } from './gitignore-writer.js';
 
-const NPM_REGISTRY = 'https://registry.npmjs.com';
+let currentRegistry: string | undefined;
+
+function getRegistry(): string {
+  if (!currentRegistry) {
+    if (process.env.npm_config_registry) {
+      currentRegistry = process.env.npm_config_registry;
+    } else if (process.env.npm_command) {
+      // if running inside npm and not having the registry than it's the default one
+      currentRegistry = 'https://registry.npmjs.com';
+    } else {
+      // Using global as when trying to get the registry inside npm workspace it would fail
+      const registry = execaSync('npm', ['config', 'get', '--global', 'registry'], {
```

Comment:
Please also add a `try ... catch` here, because a lot of stuff can go wrong here. If an error occurs we should log a warning (using the logger that you can inject) and use the default value,